libcodechecker/analyze/log_parser.py

```diff
@@ -47,75 +125,50 @@ def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
         " " + sysroot + " - -v "
 
     LOG.debug("Retrieving default includes via '" + cmd + "'")
-    include_paths = []
-    try:
-        proc = subprocess.Popen(shlex.split(cmd),
-                                stdin=subprocess.PIPE,
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-
-        _, err = proc.communicate("")
-
-        do_append = False
-        for line in err.splitlines(True):
-            line = line.strip()
-            if line.startswith(end_mark):
-                do_append = False
-            if do_append:
-                # On OSX there are framework includes,
-                # where we need to strip the "(framework directory)" string.
-                # For instance:
-                # /System/Library/Frameworks (framework directory)
-                fpos = line.find("(framework directory)")
-                if fpos == -1:
-                    include_paths.append("-isystem " + line)
-                else:
-                    include_paths.append("-isystem " + line[0:fpos-1])
-
-            if line.startswith(start_mark):
-                do_append = True
-
-    except OSError as oerr:
-        LOG.error("Cannot find include paths: " + oerr.strerror + "\n")
-    return include_paths
+    err = get_compiler_err(cmd)
+    if output_path is not None:
+        dump_compiler_info(output_path, compiler_includes_dump_file,
+                           {"compiler": compiler, "includes": err})
+    return parse_compiler_includes(err)
 
 
-def get_compiler_target(compiler):
+def get_compiler_target(compiler, output_path=None):
     """
     Returns the target triple of the given compiler as a string.
-
     """
-    target_label = "Target:"
-    target = ""
 
     cmd = compiler + ' -v'
     LOG.debug("Retrieving target platform information via '" + cmd + "'")
 
-    try:
-        proc = subprocess.Popen(shlex.split(cmd),
-                                stdin=subprocess.PIPE,
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-
-        _, err = proc.communicate("")
-        for line in err.splitlines(True):
-            line = line.strip().split()
-            if line[0] == target_label:
-                target = line[1]
+    err = get_compiler_err(cmd)
+    if output_path is not None:
+        dump_compiler_info(output_path, compiler_target_dump_file,
+                           {"compiler": compiler, "target": err})
+    return parse_compiler_target(err)
 
-    except OSError as oerr:
-        LOG.error("Cannot find compiler target: " + oerr.strerror + "\n")
 
-    return target
+def remove_file_if_exists(filename):
+    try:
+        os.remove(filename)  # Throws if file does not exsist.
```

Comment:
These comments are unneccessary here. Also, I recall we use the following form almost everywhere in the code.

```python
if os.path.isfile(a):  # Or exists()
  os.remove(a)
```